routing/index_graph_starter.cpp

```diff
@@ -189,6 +189,15 @@ void IndexGraphStarter::RedressRoute(vector<Joint::Id> const & route,
     Joint::Id const prevJoint = route[i];
     Joint::Id const nextJoint = route[i + 1];
 
+    // If a route goes through a loop-edge starting from a joint and finishing
+    // with the same joint it's necessary to skip repeated joints.
```

Comment:
It would be better to put explicit check for joint::id duplicates, then trust implicit loop below handle such use case correctly.